### PR TITLE
fix test regression

### PR DESCRIPTION
**Pull request by anamnavi**
# PR Summary

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
Add description to the PR explaining fixes

File: src/code/LocalServerApiCalls.cs [hunk 1]
```diff
@@ -1083,6 +1084,35 @@ private string[] GetCmdsOrDSCTags(string[] tags, bool isSearchingForCommands)
             return cmdDSCTags.ToArray();
         }
 
+        private void GetMetadataFilesFromPath(string dirPath, string packageName, out string psd1FilePath, out string ps1FilePath, out string nuspecFilePath)
```
Comment (alerickson) [inline src/code/LocalServerApiCalls.cs]: Wondering if this should be moved under the utils file?

File: src/code/V2ResponseUtil.cs [hunk 2]
```diff
@@ -2,8 +2,10 @@
 // Licensed under the MIT License.
 
 using Microsoft.PowerShell.PSResourceGet.UtilClasses;
+using NuGet.Versioning;
 using System;
 using System.Collections.Generic;
+using System.Linq;
```
Comment (alerickson) [inline src/code/V2ResponseUtil.cs:8]: Is this directive still needed?

File: src/code/V2ResponseUtil.cs [hunk 3]
```diff
@@ -80,7 +85,48 @@ public XmlNode[] ConvertResponseToXML(string httpResponse) {
...
+                                {
+                                    // later version element
+                                    lastVersion = parsedNormalizedVersion;
+                                }
+
+                                break; // don't care about rest of the childNode's properties
+                            }
+                        }
+                    }
+                }
```
Comment (adityapatwardhan) [inline src/code/V2ResponseUtil.cs:126]: Not sure about the XML structure, but can we also break after inspecting the "properties"?

File: src/code/V2ResponseUtil.cs [hunk 4]
```diff
@@ -80,7 +85,48 @@ public XmlNode[] ConvertResponseToXML(string httpResponse) {
...
+            }
+
+            // only order the array in desc order if array has more than 1 element and is currently in ascending order
+            // check for emptyVersion is in case a version that couldn't be parsed was found, just keep ordering as is.
+            if (nodes.Length > 1 && firstVersion != emptyVersion && lastVersion != emptyVersion && firstVersion < lastVersion)
```
Comment (adityapatwardhan) [inline src/code/V2ResponseUtil.cs]: Is it possible to publish a module with version 0.0.0.0? If so, this condition will be a problem

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: src/code/V2ResponseUtil.cs (hunk 2 above)
Comment (anamnavi) [reply src/code/V2ResponseUtil.cs:8]: yes, for the `.Reverse()`

File: src/code/V2ResponseUtil.cs (hunk 4 above)
Comment (anamnavi) [reply src/code/V2ResponseUtil.cs]: changed the condition

**Pull request closed without merging**